### PR TITLE
Attn: Chris :: Study Visit Tests

### PR DESCRIPTION
**Pull request by swashcap**
Asana task: https://app.asana.com/0/37131652076106/37607816578216

This adds tests to ensure a study’s visits function properly. It tests the creation and editing of a singular visit and ensures the data is persisted in the visits list table. There are some minor changes to pre-existing code. Also, this is good PR practice for me. If you see anything that could use improvement, let me know!

Writing these tests resulted in a minor change to MICIS (MRN-Code/micis/#333). Running the tests is dependent on that change.

### CODE REVIEW

File: test/lib/study.js [hunk 1]
```diff
@@ -55,6 +55,110 @@ module.exports = function(client, config) {
...
+         *       {
+         *           data: {
+         *               label: 'Test Label 2',
+         *               timeFromBaseline: 2,
+         *               timeUnit: 'Month'
+         *           },
+         *           mode: 'update'
+         *       }
+         *
+         * @return {Object}         Instance of `client`
```
Comment (cdaringe) [inline test/lib/study.js]: no rework req'd.  we recently decided to lowercase primatives, FYI. not sure if that made it back int our docs.  sexy doc blocks!

File: test/lib/study.js [hunk 2]
```diff
@@ -55,6 +55,110 @@ module.exports = function(client, config) {
...
+         */
+        fillOutForm: function (options) {
+            if (! options.data) {
```
Comment (cdaringe) [inline test/lib/study.js]: `!_options.data` ==> `!options.data`.

FYI, I still need to integrate our jscs into a pre-commit hook, using https://github.com/nlf/precommit-hook .  we have our standard here https://github.com/MRN-Code/javascriptStyleguide, i just haven't pieced it all together.  once we get that dialed, integrate it into https://github.com/MRN-Code/node_boilerplate

If you want to help out with that at some point, you're welcome to

File: test/lib/study.js [hunk 3]
```diff
@@ -55,6 +55,110 @@ module.exports = function(client, config) {
...
+                });
+        },
+        navigateToEditPage: function (segmentInterval) {
+            var selector = '//tr/td[. = "' + segmentInterval + '"]/following-sibling::td/a';
+            return client
+                .scroll(selector, 0, -30, function (err) {
+                    if (err) {
+                        throw err;
+                    }
+                })
```
Comment (cdaringe) [inline test/lib/study.js]: what's goin on here?

Comment (swashcap) [inline test/lib/study.js]: Expression split to bring that line length down: 0e5dd4e5b02b14693330bb6ee0d6a0cdccda9981

Comment (cdaringe) [inline test/lib/study.js]: sorry, i meant to ask "why do you need a manual scroll?".  just to get it in the view?

**Pull request merged**